pkg/host/nvidia.go

```diff
@@ -0,0 +1,64 @@
+package host
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+
+	"github.com/elotl/itzo/pkg/util"
+	"github.com/golang/glog"
+)
+
+const (
+	NvidiaContainerCli = "nvidia-container-cli"
+	NvidiaSMI          = "nvidia-smi"
+)
+
+// See https://nvidia.github.io/libnvidia-container/ on how to install
+// nvidia-container-cli. We add it to our Ubuntu-based images.
+func InitializeGPU(rootfs string) error {
+	if _, err := os.Stat("/dev/nvidiactl"); err != nil {
+		if os.IsNotExist(err) {
+			// Not a GPU instance.
+			return nil
+		}
+		glog.Errorf("Checking /dev/nvidiactl: %v", err)
+		return err
+	}
+	cli, err := util.EnsureProg(NvidiaContainerCli, "https://s3.amazonaws.com/itzo-packages/nvidia-container-cli", "1.0.2", "--version")
```

Comment:
Just a note for a future refactor, probably doesn't need to happen now: If we start downloading more and more stuff, we might want to move all of these URL strings to their own file so we can bulk update them if needed.  We might want to distribute these files across multiple clouds (but that's a bit more involved)?

Reply:
Good point. I'm on the fence whether we need a backup mechanism like that at all, or just focus on building a robust image build pipeline that ensures all required packages/programs are present.

For now, since there might be users out there using an old image but requesting the latest version of itzo, I think it's better to keep this in place. I'll move the URL to a const.